action/protocol/staking/protocol.go

```diff
@@ -442,7 +466,12 @@ func (p *Protocol) ReadState(ctx context.Context, sr protocol.StateReader, metho
 	case iotexapi.ReadStakingDataMethod_CANDIDATE_BY_ADDRESS:
 		resp, height, err = readStateCandidateByAddress(ctx, csr, r.GetCandidateByAddress())
 	case iotexapi.ReadStakingDataMethod_TOTAL_STAKING_AMOUNT:
-		resp, height, err = readStateTotalStakingAmount(ctx, csr, r.GetTotalStakingAmount())
+		if p.archiveMode && p.hu.IsPre(config.Greenland, height) {
+			resp, height, err = readStateTotalStakingAmountFromHeight(csr, r.GetTotalStakingAmount(), inputHeight)
```

Comment:
`readStateTotalStakingAmountFromHeight` to `readStateTotalStakingAmountFromIndexer`